Automattic/jetpack
# Move "All Sites" link to the top bar

_@okmttdhr took over this PR_

## Proposed changes:


This PR removes "All Sites" from the sidebar and adds "All Sites" to the top bar in wp-admin view. 

| before | after |
|--------|--------|
| ![Image](https://github.com/Automattic/dotcom-forge/assets/5634774/adb50127-1cd5-415d-b4a4-da81285ba52b) | <img width="425" alt="Screenshot 2024-04-08 at 18 06 43" src="https://github.com/Automattic/jetpack/assets/5287479/6cd814c3-0300-49b6-befa-5220423aaea2"> <img width="304" alt="Screenshot 2024-04-08 at 17 56 41" src="https://github.com/Automattic/jetpack/assets/5287479/68181bf3-06e1-42e9-8ffa-c757166ea6ca">|

- Part of https://github.com/Automattic/dotcom-forge/issues/6262
- Related to 
	- https://github.com/Automattic/wp-calypso/pull/89201
	- https://github.com/Automattic/wp-calypso/pull/89321
	- https://github.com/Automattic/jetpack/pull/36341
	- D143420-code
- UI p9Jlb4-btR-p2

### How

#### SVG

- According to p9Jlb4-btR-p2, the task requires the use of the category icon from the `category` icon available at https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library. 
- Given the absence of an equivalent icon in the [dashicon](https://developer.wordpress.org/resource/dashicons) library, incorporates raw SVG data directly from the Gutenberg repository. 

#### Applying color scheme


See https://github.com/Automattic/jetpack/pull/36632#discussion_r1557015871

<details><summary>previous approach</summary>
<p>

#### For Core's color schemes (Default(Fresh), Light, Modern, Blue, Coffee, Ectoplasm, Midnight, Ocean, Sunrise)
I introduced new CSS files for each color scheme. 
(Note that this is not needed if there is an equivalent dashicon, as we can just use the default styles from Core. Also, the file: `admin-color-schemes/colors/{color}/colors.scss` was not suitable for our needs due to inconsistencies with Core's color definitions). 

#### For dotcom-specific color schemes 
I added styles in `admin-color-schemes/colors/_overrides.scss` to match dotcom styles. 

</p>
</details> 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






- Prepare an Atomic site.
- Go to Hosting Configuration and set Admin interface style to Classic.
- Go to the site's _cli and run wp update option `wpcom_classic_early_release` 1.
- Patch this PR via https://github.com/Automattic/jetpack/pull/36632#issuecomment-2024745978
- Go to `/wp-admin` 
- Ensure "All Sites" menu redirects you to `wp.com/sites`
- Change the color scheme
- Ensure "All Sites" menu looks consistent with other menu items
